sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/ContentType.java

```diff
@@ -32,4 +32,21 @@ public final class ContentType {
      */
     private ContentType() {
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        ContentType that = (ContentType) obj;
+        return this.toString().toLowerCase().startsWith(that.toString().toLowerCase());
+    }
```

Comment:
I don't think we should change the `equals()` method to use `startsWith`. The contract for overriding equals is that if two objects are `equal`, they should be symmetric and have the same `hashcode` too. 

Symmetry:
`"application/json".equals("application/json; charset=utf8")` is not the same as `"application/json; charset=utf8".equals("application/json")`

Hashcode:
`application/json` and `application/json; charset=utf8` are considered equal but they don't have the same hashcode.

Reply:
Thoughts on dropping `equals` and `hashCode` entirely here and where this was being used `String.regionMatches` can be used to perform a more optimized check to make sure we found what we're looking for?

Reply:
Also FWIW, this doesn't even work as `ContentType` can't be instantiated and the constants on the class are Strings.